pkg/models/monitoring/monitoring.go

```diff
@@ -38,14 +40,22 @@ func NewMonitoringOperator(client monitoring.Interface) MonitoringOperator {
 	return &monitoringOperator{client}
 }
 
-// TODO(huanggze): reserve for custom monitoring
-func (mo monitoringOperator) GetMetrics(stmts []string, time time.Time) Metrics {
-	panic("implement me")
+func (mo monitoringOperator) GetMetric(expr, namespace string, time time.Time) (monitoring.Metric, error) {
+	// Enforce namespace matcher
+	expr, err := expressions.ReplaceNamespaceFns["prometheus"](expr, namespace)
```

Comment:
Is it better to check if the value of key "prometheus" exist first, and then call the label replace function?

Reply:
The idea is that we allow for various monitoring backend implements, and prometheus is just one of them. Different implements have different ways to enforce namespace isolation. No matter how, all of them should register themselves in `ReplaceNamespaceFns` during init. 

Below is how prometheus conducts. So the key "prometheus" always exists. We hard code here because we only have one monitoring backend imeplemets so far, that is prometheus. But the framework allows extension.
 
https://github.com/kubesphere/kubesphere/commit/b09541729a84d99edecf7e21e29965c54c76a152#diff-791d0597894092e1fe80cdaee06c798eR11-R13

Reply:
I see. It's ok as long as no nil reference occurs.

Reply:
I put more comments on this issue for further clarification.

Reply:
That's good.
Two minor syntax issues.